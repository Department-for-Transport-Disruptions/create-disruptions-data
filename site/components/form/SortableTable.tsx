import { SortOrder } from "@create-disruptions-data/shared-ts/enums";
import { Dispatch, ReactElement, ReactNode, SetStateAction, useEffect, useState } from "react";
import PageNumbers from "../layout/PageNumbers";

export interface TableColumn<T> {
    displayName: string;
    key: keyof T;
    sortable?: boolean;
    widthClass?: string;
}

interface SortableTableProps<T> {
    columns: TableColumn<T>[];
    rows: T[];
    pageCount: number;
    setCurrentPage: Dispatch<SetStateAction<number>>;
    currentPage: number;
    caption?: { text: string; size: "s" | "m" | "l" };
    sortOrder: SortOrder;
    setSortOrder: Dispatch<SetStateAction<SortOrder>>;
    sortedField: keyof T | null;
    setSortedField: Dispatch<SetStateAction<keyof T | null>>;
}

const SortableTable = <T extends object>({
    columns,
    rows,
    caption,
    pageCount,
    setCurrentPage,
    currentPage,
    sortedField,
    setSortedField,
    sortOrder,
    setSortOrder,
}: SortableTableProps<T>): ReactElement => {
    const [displayRows, setDisplayRows] = useState<T[]>(rows);

    const sortData = (field: keyof T) => {
        if (sortedField === field) {
            setSortOrder(sortOrder === SortOrder.asc ? SortOrder.desc : SortOrder.asc);
        } else {
            setSortedField(field);
            setSortOrder(SortOrder.asc);
        }
    };

    useEffect(() => {
        setDisplayRows(rows);
    }, [rows]);

    const getArrowHeads = (field: keyof T) => {
        if (!sortedField || sortedField !== field) {
            return (
                <div className="flex flex-col pl-1 text-arrow">
                    <span className="-mb-1">&#x25B2;</span>
                    <span>&#x25BC;</span>
                </div>
            );
        }
        return sortOrder === SortOrder.asc ? (
            <span className="pl-1 govuk-!-font-size-14">&#x25B2;</span>
        ) : (
            <span className="pl-1 govuk-!-font-size-14">&#x25BC;</span>
        );
    };

    return (
        <>
            <table className="govuk-table">
                {caption ? (
                    <caption className={`govuk-table__caption govuk-table__caption--${caption.size}`}>
                        {caption.text}
                    </caption>
                ) : (
                    ""
                )}
                <thead className="govuk-table__head">
                    <tr className="govuk-table__row">
                        {columns.map((column) => {
                            return column.sortable ? (
                                <th
                                    scope="col"
                                    className={`govuk-table__header align-middle px-2 ${
                                        column.widthClass ? column.widthClass : ""
                                    }`}
                                    key={`header-${column.displayName}`}
                                    onClick={() => sortData(column.key)}
                                >
                                    <div className="flex flex-row items-center">
                                        {column.displayName}
                                        {getArrowHeads(column.key)}
                                    </div>
                                </th>
                            ) : (
                                <th
                                    scope="col"
                                    className={`govuk-table__header align-middle px-2 ${
                                        column.widthClass ? column.widthClass : ""
                                    }`}
                                    key={`header-${column.displayName}`}
                                >
                                    {column.displayName}
                                </th>
                            );
                        })}
                    </tr>
                </thead>
                <tbody className="govuk-table__body">
                    {displayRows.map((row, index) => (
                        <tr className="govuk-table__row" key={`row-${index}`}>
                            {columns.map((column, i) => (
                                <td className={"govuk-table__cell align-middle px-2"} key={`row-${index}${i}`}>
                                    {row[column.key] as ReactNode}
                                </td>
                            ))}
                        </tr>
                    ))}
                </tbody>
            </table>
            <PageNumbers numberOfPages={pageCount} currentPage={currentPage} setCurrentPage={setCurrentPage} />
        </>
    );
};

export default SortableTable;
