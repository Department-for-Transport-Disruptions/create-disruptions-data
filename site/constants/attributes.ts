export const CD_SUMMARY = "summary";

export const CD_DESCRIPTION = "description";

export const CD_DISRUPTION_TYPE = "disruptionType";

export const CD_ASSOCIATED_LINK = "associatedLink";

export const CD_DISRUPTION_REASON = "disruptionReason";

export const CD_DISRUPTION_REAPEATS = "disruptionRepeats";

export const CD_NO_VALIDITY_END_DATE_TIME = "validityNoEndDateTime";

export const CD_NO_PUBLISH_END_DATE_TIME = "publishNoEndDateTime";

export const CD_VALIDITY_START_DATE_DAY = "validityStartDateDay";

export const CD_VALIDITY_START_DATE_MONTH = "validityStartDateMonth";

export const CD_VALIDITY_START_DATE_YEAR = "validityStartDateYear";

export const CD_VALIDITY_START_HOUR = "validityStartHour";

export const CD_VALIDITY_START_MINUTES = "validityStartMinute";

export const CD_VALIDITY_END_DATE_DAY = "validityEndDateDay";

export const CD_VALIDITY_END_DATE_MONTH = "validityEndDateMonth";

export const CD_VALIDITY_END_DATE_YEAR = "validityEndDateYear";

export const CD_VALIDITY_END_HOUR = "validityEndHour";

export const CD_VALIDITY_END_MINUTES = "validityEndMinute";

export const CD_PUBLISH_START_DATE_DAY = "publishStartDateDay";

export const CD_PUBLISH_START_DATE_MONTH = "publishStartDateMonth";

export const CD_PUBLISH_START_DATE_YEAR = "publishStartDateYear";

export const CD_PUBLISH_START_HOUR = "publishStartHour";

export const CD_PUBLISH_START_MINUTES = "publishStartMinute";

export const CD_PUBLISH_END_DATE_DAY = "publishEndDateDay";

export const CD_PUBLISH_END_DATE_MONTH = "publishEndDateMonth";

export const CD_PUBLISH_END_DATE_YEAR = "publishEndDateYear";

export const CD_PUBLISH_END_HOUR = "publishEndHour";

export const CD_PUBLISH_END_MINUTES = "publishEndMinute";
