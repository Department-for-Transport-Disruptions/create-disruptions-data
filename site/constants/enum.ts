export enum TransportMode {
    bus = "Bus",
    tram = "Tram",
    ferry = "Ferry",
    train = "Train",
}

export enum ConsequenceType {
    services = "Services",
    networkWide = "Network wide",
    operatorWide = "Operator wide",
    stops = "Stops",
}
