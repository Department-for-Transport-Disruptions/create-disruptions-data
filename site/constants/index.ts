import {
    MiscellaneousReason,
    PersonnelReason,
    EnvironmentReason,
    EquipmentReason,
} from "@create-disruptions-data/shared-ts/enums";

export const SUPPORT_EMAIL_ADDRESS = "bodshelpdesk@kpmg.co.uk";

export const SERVICE_EMAIL_ADDRESS = "bodshelpdesk@kpmg.co.uk";

export const SUPPORT_PHONE_NUMBER = "0800 123 1234";

export const DISRUPTION_REASONS: {
    value: MiscellaneousReason | PersonnelReason | EnvironmentReason | EquipmentReason;
    display: string;
}[] = [
    {
        value: MiscellaneousReason.roadWorks,
        display: "Road Works",
    },
    {
        value: MiscellaneousReason.vandalism,
        display: "Vandalism",
    },
    {
        value: MiscellaneousReason.routeDiversion,
        display: "Route Diversion",
    },
    {
        value: MiscellaneousReason.specialEvent,
        display: "Special Event",
    },
];

export const DISRUPTION_TYPES = ["planned", "unplanned"];

export const CD_DATE_FORMAT = "DD/MM/YYYY";
export const CONTACT_FEEDBACK_QUESTION = "Did you contact us for assistance at any point?";
export const SOLVE_FEEDBACK_QUESTION = "Did we solve your problem?";
export const HEAR_ABOUT_US_FEEDBACK_QUESTION = "How did you hear about our service?";
export const GENERAL_FEEDBACK_QUESTION = "Please let us know any feedback or suggestions for improvements you may have";

export const COOKIES_POLICY_COOKIE = "cdd-cookies-policy";

export const COOKIE_PREFERENCES_COOKIE = "cdd-cookie-preferences-set";

export const oneYearInSeconds = 31556952;

export const ID_TOKEN_COOKIE = "cdd-id-token";

export const COOKIES_DISRUPTION_INFO = "disruption-info";

export const COOKIES_DISRUPTION_ERRORS = "disruption-errors";

export const COOKIES_ADD_CONSEQUENCE_INFO = "cdd-consequence-info";

export const COOKIES_ADD_CONSEQUENCE_ERRORS = "cdd-consequence-errors";

export const CREATE_DISRUPTION_PAGE_PATH = "/create-disruption";

export const ERROR_PATH = "/_error";

export const ADD_CONSEQUENCE_PAGE_PATH = "/type-of-consequence";

export const STAGE = process.env.STAGE || "dev";
