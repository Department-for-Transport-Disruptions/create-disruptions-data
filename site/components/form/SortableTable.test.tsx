import Link from "next/link";
import renderer from "react-test-renderer";
import { describe, it, expect } from "vitest";
import SortableTable, { TableColumn } from "./SortableTable";
import { DISRUPTION_DETAIL_PAGE_PATH } from "../../constants";

interface RandomTable {
    id: string | JSX.Element;
    name: string;
    start: string;
    role: string;
}
const defaultColumn: TableColumn<RandomTable>[] = [
    {
        displayName: "ID",
        key: "id",
    },
    {
        displayName: "Name",
        key: "name",
    },
    {
        displayName: "Start",
        key: "start",
        sortable: true,
    },
    {
        displayName: "Role",
        key: "role",
        sortable: true,
    },
];

const defaultData: RandomTable[] = [
    {
        id: (
            <Link
                className="govuk-link"
                href={{
                    pathname: `${DISRUPTION_DETAIL_PAGE_PATH}/id`,
                }}
                key="key-1"
            >
                aft62h
            </Link>
        ),
        name: "Random name1",
        start: "25/03/2023",
        role: "Engineer",
    },
    {
        id: (
            <Link
                className="govuk-link"
                href={{
                    pathname: `${DISRUPTION_DETAIL_PAGE_PATH}/id`,
                }}
                key="key-2"
            >
                aft62h
            </Link>
        ),
        name: "Random name4",
        start: "25/03/2023",
        role: "Architect",
    },
    {
        id: (
            <Link
                className="govuk-link"
                href={{
                    pathname: `${DISRUPTION_DETAIL_PAGE_PATH}/id`,
                }}
                key="key-3"
            >
                aft62h
            </Link>
        ),
        name: "Random name2",
        start: "20/04/2023",
        role: "Engineer",
    },
    {
        id: (
            <Link
                className="govuk-link"
                href={{
                    pathname: `${DISRUPTION_DETAIL_PAGE_PATH}/id`,
                }}
                key="key-1"
            >
                aft62h
            </Link>
        ),
        name: "Random name3",
        start: "25/03/2024",
        role: "Director",
    },
];

const defaultSortFunction = (rows: RandomTable[], sortField: keyof RandomTable) => {
    return rows.sort((a, b) => {
        return a[sortField].toString().localeCompare(b[sortField].toString());
    });
};

describe("SortableTable", () => {
    it("should render correctly", () => {
        const tree = renderer.create(<SortableTable columns={defaultColumn} rows={defaultData} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly when empty arrays are passed", () => {
        const tree = renderer.create(<SortableTable columns={[]} rows={[]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly with pagination", () => {
        const tree = renderer
            .create(
                <SortableTable
                    columns={defaultColumn}
                    rows={Array.from({ length: 10 }, () => [...defaultData]).flat()}
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly with sorting function", () => {
        const tree = renderer
            .create(
                <SortableTable
                    columns={defaultColumn}
                    rows={Array.from({ length: 10 }, () => [...defaultData]).flat()}
                    sortFunction={defaultSortFunction}
                />,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
