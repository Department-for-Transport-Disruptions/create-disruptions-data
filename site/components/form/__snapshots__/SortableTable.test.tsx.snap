// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SortableTable > should render correctly 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          ID
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          Name
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          <div
            class="flex flex-row items-center"
          >
            Start
            <div
              class="flex flex-col pl-1 text-arrow"
            >
              <span
                class="-mb-1"
              >
                ▲
              </span>
              <span>
                ▼
              </span>
            </div>
          </div>
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          <div
            class="flex flex-row items-center"
          >
            Role
            <div
              class="flex flex-col pl-1 text-arrow"
            >
              <span
                class="-mb-1"
              >
                ▲
              </span>
              <span>
                ▼
              </span>
            </div>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name1
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Engineer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name4
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Architect
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name2
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          20/04/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Engineer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name3
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2024
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Director
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`SortableTable > should render correctly when empty arrays are passed 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      />
    </thead>
    <tbody
      class="govuk-table__body"
    />
  </table>
</DocumentFragment>
`;

exports[`SortableTable > should render correctly with pagination 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          ID
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          Name
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          <div
            class="flex flex-row items-center"
          >
            Start
            <div
              class="flex flex-col pl-1 text-arrow"
            >
              <span
                class="-mb-1"
              >
                ▲
              </span>
              <span>
                ▼
              </span>
            </div>
          </div>
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          <div
            class="flex flex-row items-center"
          >
            Role
            <div
              class="flex flex-col pl-1 text-arrow"
            >
              <span
                class="-mb-1"
              >
                ▲
              </span>
              <span>
                ▼
              </span>
            </div>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name1
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Engineer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name4
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Architect
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name2
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          20/04/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Engineer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name3
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2024
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Director
        </td>
      </tr>
    </tbody>
  </table>
  <nav
    class="govuk-pagination justify-center"
  >
    <ul
      class="govuk-pagination__list"
    >
       
      <li
        class="govuk-pagination__item govuk-pagination__item--current"
      >
        <a
          aria-label="Page 1"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          1 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 2"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          2 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 3"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          3 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 4"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          4 
        </a>
      </li>
    </ul>
    <div
      class="govuk-pagination__next"
    >
      <a
        aria-label="Page 4"
        class="govuk-link govuk-pagination__link"
        data-module="govuk-button"
        draggable="false"
        href=""
        rel="next"
        role="button"
      >
        <span
          class="govuk-pagination__link-title"
        >
          Next
        </span>
         
        <svg
          aria-hidden="true"
          class="govuk-pagination__icon govuk-pagination__icon--next"
          focusable="false"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"
          />
        </svg>
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`SortableTable > should render correctly with sorted field 1`] = `
<DocumentFragment>
  <table
    class="govuk-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row"
      >
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          ID
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          Name
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          <div
            class="flex flex-row items-center"
          >
            Start
            <span
              class="pl-1 govuk-!-font-size-14"
            >
              ▲
            </span>
          </div>
        </th>
        <th
          class="govuk-table__header align-middle px-2 "
          scope="col"
        >
          <div
            class="flex flex-row items-center"
          >
            Role
            <div
              class="flex flex-col pl-1 text-arrow"
            >
              <span
                class="-mb-1"
              >
                ▲
              </span>
              <span>
                ▼
              </span>
            </div>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name1
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Engineer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name4
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Architect
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name2
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          20/04/2023
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Engineer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
      >
        <td
          class="govuk-table__cell align-middle px-2"
        >
          <a
            class="govuk-link"
            href="/disruption-detail/id?return=%2Fview-all-disruptions"
          >
            aft62h
          </a>
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Random name3
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          25/03/2024
        </td>
        <td
          class="govuk-table__cell align-middle px-2"
        >
          Director
        </td>
      </tr>
    </tbody>
  </table>
  <nav
    class="govuk-pagination justify-center"
  >
    <ul
      class="govuk-pagination__list"
    >
       
      <li
        class="govuk-pagination__item govuk-pagination__item--current"
      >
        <a
          aria-label="Page 1"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          1 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 2"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          2 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 3"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          3 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 4"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          4 
        </a>
      </li>
    </ul>
    <div
      class="govuk-pagination__next"
    >
      <a
        aria-label="Page 4"
        class="govuk-link govuk-pagination__link"
        data-module="govuk-button"
        draggable="false"
        href=""
        rel="next"
        role="button"
      >
        <span
          class="govuk-pagination__link-title"
        >
          Next
        </span>
         
        <svg
          aria-hidden="true"
          class="govuk-pagination__icon govuk-pagination__icon--next"
          focusable="false"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"
          />
        </svg>
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
